### DISCUSSION TITLE
How to use "STR" in Block mode

### DISCUSSION THREAD

**Discussion by didiersm** [pybricks/support#1323] [General]
I just want to broadcast data, but I need STR function in Block Mode!
HowTo?
Thanks

**Comment by laurensvalk** [comment]
Can you describe your use case? It may be possible with the existing blocks.

There is a block for a string literal, on the data category. You can also concatenate strings by using +, just like Python.

**Reply by didiersm** [reply]
![looking for Str](https://github.com/pybricks/support/assets/75566618/f8a587e7-e407-4bcd-b3c8-ef8f4abe52eb)

**Reply by laurensvalk** [reply]
Ah, you'd like to broadcast the color.

I've been thinking to add a generic format block with a variable number of inputs, and this use case could be covered by that as well. 

But even if broadcast as a string, you'd have to do string comparisons on the other side to convert it back to something you can use.

So another thing you could do is define a function that converts the color to a numeric index and broadcast that. I'll see if I can put together an example. 

For colors specifically, another thing we could do is make them iterable for h, s, v, (in the firmware) which can also be useful for other applications. The blocks have a way to unpack iterables already, so this would then let you broadcast the numeric h, s, hue values.

**Reply by didiersm** [reply]
Thanks for your answer, if you have an example it will be appreciate ;)
I have found a roundabout If Color.RED I broadcast 1 and for the YELLOW 2 and that's ok.
Another question: I remark that the Broadcast function never stop to broadcast, How can I stop it?

**Reply by laurensvalk** [reply]
Here is one way to encode the colors:

![image](https://github.com/pybricks/support/assets/12326241/fe870dc8-3009-444f-801b-81b9c8d0112d)

Alternatively, put all your colors in a list, and change the function to find the index that has the same value.

> Another question: I remark that the Broadcast function never stop to broadcast, How can I stop it?

We could consider allowing broadcasting to stop if you broadcast `None`, but this is not done yet.

**Reply by rbeckett75** [reply]
this is similar to what I wanted to do in one of my projects:  I'm building a GBC train, and would like the train to communicate with the hub at each end, sending a 'load' signal... with the loader hub sending a reply back to the train when done "done loading - travel to other station"

I don't need to send continuous signals, but rather a simple command that is actioned on, when recieved.

**Reply by laurensvalk** [reply]
> I don't need to send continuous signals, but rather a simple command that is actioned on, when recieved.

Due to the way broadcasting works (receive only if it happens to listen), it's often better to broadcast all the time or at least for some sustained time.

So as opposed to "load", you might want to send "do load"/"not load" (or just `True`/`False` is that's all you need) so you can distinguish the state from receiving nothing at all.
